diff --git a/XenosRecomp/CMakeLists.txt b/XenosRecomp/CMakeLists.txt
index e505a5e..0574517 100644
--- a/XenosRecomp/CMakeLists.txt
+++ b/XenosRecomp/CMakeLists.txt
@@ -28,7 +28,7 @@ target_include_directories(XenosRecomp PRIVATE ${SMOLV_SOURCE_DIR})

 target_precompile_headers(XenosRecomp PRIVATE pch.h)

-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
     target_compile_options(XenosRecomp PRIVATE -Wno-switch -Wno-unused-variable -Wno-null-arithmetic -fms-extensions)

     include(CheckCXXSymbolExists)
diff --git a/XenosRecomp/main.cpp b/XenosRecomp/main.cpp
index d015145..374ea11 100644
--- a/XenosRecomp/main.cpp
+++ b/XenosRecomp/main.cpp
@@ -9,7 +9,7 @@ static std::unique_ptr<uint8_t[]> readAllBytes(const char* filePath, size_t& fil
     fileSize = ftell(file);
     fseek(file, 0, SEEK_SET);
     auto data = std::make_unique<uint8_t[]>(fileSize);
-    fread(data.get(), 1, fileSize, file);
+    if (fread(data.get(), 1, fileSize, file) != fileSize) { fclose(file); return nullptr; }
     fclose(file);
     return data;
 }
diff --git a/XenosRecomp/shader_recompiler.cpp b/XenosRecomp/shader_recompiler.cpp
index 698a418..047b0d5 100644
--- a/XenosRecomp/shader_recompiler.cpp
+++ b/XenosRecomp/shader_recompiler.cpp
@@ -1403,13 +1403,13 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
                         int8_t y;
                         int8_t z;
                         int8_t w;
-                    };
-                };
+                    } raw;
+                } u;

-                value = definition->values[i].get();
+                u.value = definition->values[i].get();

                 println("\tint4 i{} = int4({}, {}, {}, {});",
-                    (definition->registerIndex - 8992) / 4 + i, x, y, z, w);
+                    (definition->registerIndex - 8992) / 4 + i, u.raw.x, u.raw.y, u.raw.z, u.raw.w);
             }
             definitions += 2;
             definitions += definition->count;
@@ -1502,8 +1502,8 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
             uint32_t code1;
             uint32_t code2;
             uint32_t code3;
-        };
-    };
+        } raw;
+    } u;

     auto controlFlowCode = code;
     uint32_t instrAddress = 0;
@@ -1512,12 +1512,12 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi

     while (instrAddress < instrSize)
     {
-        code0 = controlFlowCode[0];
-        code1 = controlFlowCode[1] & 0xFFFF;
-        code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
-        code3 = controlFlowCode[2] >> 16;
+        u.raw.code0 = controlFlowCode[0];
+        u.raw.code1 = controlFlowCode[1] & 0xFFFF;
+        u.raw.code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
+        u.raw.code3 = controlFlowCode[2] >> 16;

-        for (auto& cfInstr : controlFlow)
+        for (auto& cfInstr : u.controlFlow)
         {
             uint32_t address = 0;

@@ -1579,12 +1579,12 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi

     while (instrAddress < instrSize)
     {
-        code0 = controlFlowCode[0];
-        code1 = controlFlowCode[1] & 0xFFFF;
-        code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
-        code3 = controlFlowCode[2] >> 16;
+        u.raw.code0 = controlFlowCode[0];
+        u.raw.code1 = controlFlowCode[1] & 0xFFFF;
+        u.raw.code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
+        u.raw.code3 = controlFlowCode[2] >> 16;

-        for (auto& cfInstr : controlFlow)
+        for (auto& cfInstr : u.controlFlow)
         {
             if (!simpleControlFlow)
             {
@@ -1733,23 +1733,23 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
                         uint32_t code0;
                         uint32_t code1;
                         uint32_t code2;
-                    };
-                };
+                    } raw;
+                } inst;

-                code0 = instructionCode[0];
-                code1 = instructionCode[1];
-                code2 = instructionCode[2];
+                inst.raw.code0 = instructionCode[0];
+                inst.raw.code1 = instructionCode[1];
+                inst.raw.code2 = instructionCode[2];

                 if ((sequence & 0x1) != 0)
                 {
-                    if (vertexFetch.opcode == FetchOpcode::VertexFetch)
+                    if (inst.vertexFetch.opcode == FetchOpcode::VertexFetch)
                     {
-                        recompile(vertexFetch, address + i);
+                        recompile(inst.vertexFetch, address + i);
                     }
                     else
                     {
                     #ifdef UNLEASHED_RECOMP
-                        if (textureFetch.constIndex == 10) // g_GISampler
+                        if (inst.textureFetch.constIndex == 10) // g_GISampler
                         {
                             specConstantsMask |= SPEC_CONSTANT_BICUBIC_GI_FILTER;

@@ -1759,7 +1759,7 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
                             out += '{';

                             ++indentation;
-                            recompile(textureFetch, true);
+                            recompile(inst.textureFetch, true);
                             --indentation;

                             indent();
@@ -1770,7 +1770,7 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
                             out += '{';

                             ++indentation;
-                            recompile(textureFetch, false);
+                            recompile(inst.textureFetch, false);
                             --indentation;

                             indent();
@@ -1779,13 +1779,13 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
                         else
                     #endif
                         {
-                            recompile(textureFetch, false);
+                            recompile(inst.textureFetch, false);
                         }
                     }
                 }
                 else
                 {
-                    recompile(alu);
+                    recompile(inst.alu);
                 }

                 sequence >>= 2;
