diff --git a/XenonRecomp/recompiler.cpp b/XenonRecomp/recompiler.cpp
index fa91527..2ccea2e 100644
--- a/XenonRecomp/recompiler.cpp
+++ b/XenonRecomp/recompiler.cpp
@@ -438,7 +438,7 @@ bool Recompiler::Recompile(
     auto printMidAsmHook = [&]()
         {
             bool returnsBool = midAsmHook->second.returnOnFalse || midAsmHook->second.returnOnTrue ||
-                midAsmHook->second.jumpAddressOnFalse != NULL || midAsmHook->second.jumpAddressOnTrue != NULL;
+                midAsmHook->second.jumpAddressOnFalse != 0 || midAsmHook->second.jumpAddressOnTrue != 0;

             print("\t");
             if (returnsBool)
@@ -489,7 +489,7 @@ bool Recompiler::Recompile(

                 if (midAsmHook->second.returnOnTrue)
                     println("\t\treturn;");
-                else if (midAsmHook->second.jumpAddressOnTrue != NULL)
+                else if (midAsmHook->second.jumpAddressOnTrue != 0)
                     println("\t\tgoto loc_{:X};", midAsmHook->second.jumpAddressOnTrue);

                 println("\t}}");
@@ -498,7 +498,7 @@ bool Recompiler::Recompile(

                 if (midAsmHook->second.returnOnFalse)
                     println("\t\treturn;");
-                else if (midAsmHook->second.jumpAddressOnFalse != NULL)
+                else if (midAsmHook->second.jumpAddressOnFalse != 0)
                     println("\t\tgoto loc_{:X};", midAsmHook->second.jumpAddressOnFalse);

                 println("\t}}");
@@ -509,7 +509,7 @@ bool Recompiler::Recompile(

                 if (midAsmHook->second.ret)
                     println("\treturn;");
-                else if (midAsmHook->second.jumpAddress != NULL)
+                else if (midAsmHook->second.jumpAddress != 0)
                     println("\tgoto loc_{:X};", midAsmHook->second.jumpAddress);
             }
         };
@@ -2314,7 +2314,7 @@ bool Recompiler::Recompile(const Function& fn)
         if (midAsmHook != config.midAsmHooks.end())
         {
             if (midAsmHook->second.returnOnFalse || midAsmHook->second.returnOnTrue ||
-                midAsmHook->second.jumpAddressOnFalse != NULL || midAsmHook->second.jumpAddressOnTrue != NULL)
+                midAsmHook->second.jumpAddressOnFalse != 0 || midAsmHook->second.jumpAddressOnTrue != 0)
             {
                 print("extern bool ");
             }
@@ -2361,11 +2361,11 @@ bool Recompiler::Recompile(const Function& fn)

             println(");\n");

-            if (midAsmHook->second.jumpAddress != NULL)
+            if (midAsmHook->second.jumpAddress != 0)
                 labels.emplace(midAsmHook->second.jumpAddress);
-            if (midAsmHook->second.jumpAddressOnTrue != NULL)
+            if (midAsmHook->second.jumpAddressOnTrue != 0)
                 labels.emplace(midAsmHook->second.jumpAddressOnTrue);
-            if (midAsmHook->second.jumpAddressOnFalse != NULL)
+            if (midAsmHook->second.jumpAddressOnFalse != 0)
                 labels.emplace(midAsmHook->second.jumpAddressOnFalse);
         }
     }
