diff --git a/XenosRecomp/shader_recompiler.cpp b/XenosRecomp/shader_recompiler.cpp
index 698a418..e9674ef 100644
--- a/XenosRecomp/shader_recompiler.cpp
+++ b/XenosRecomp/shader_recompiler.cpp
@@ -1502,7 +1502,7 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi
             uint32_t code1;
             uint32_t code2;
             uint32_t code3;
-        };
+        } codes;
     };

     auto controlFlowCode = code;
@@ -1512,10 +1512,10 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi

     while (instrAddress < instrSize)
     {
-        code0 = controlFlowCode[0];
-        code1 = controlFlowCode[1] & 0xFFFF;
-        code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
-        code3 = controlFlowCode[2] >> 16;
+        codes.code0 = controlFlowCode[0];
+        codes.code1 = controlFlowCode[1] & 0xFFFF;
+        codes.code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
+        codes.code3 = controlFlowCode[2] >> 16;

         for (auto& cfInstr : controlFlow)
         {
@@ -1579,10 +1579,10 @@ void ShaderRecompiler::recompile(const uint8_t* shaderData, const std::string_vi

     while (instrAddress < instrSize)
     {
-        code0 = controlFlowCode[0];
-        code1 = controlFlowCode[1] & 0xFFFF;
-        code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
-        code3 = controlFlowCode[2] >> 16;
+        codes.code0 = controlFlowCode[0];
+        codes.code1 = controlFlowCode[1] & 0xFFFF;
+        codes.code2 = (controlFlowCode[1] >> 16) | (controlFlowCode[2] << 16);
+        codes.code3 = controlFlowCode[2] >> 16;

         for (auto& cfInstr : controlFlow)
         {
