diff --git a/CMakeLists.txt b/CMakeLists.txt
index b89407f..53fa2af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,8 +20,10 @@ set(XENONRECOMP_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/XenonRecomp)

 project ("XenonRecomp-ALL")

-add_subdirectory(${XENONANALYSE_ROOT})
-add_subdirectory(${XENONRECOMP_ROOT})
+if (NOT ANDROID)
+    add_subdirectory(${XENONANALYSE_ROOT})
+    add_subdirectory(${XENONRECOMP_ROOT})
+endif()
 add_subdirectory(${XENONUTILS_ROOT})

 # Only tests if this is the top level project
diff --git a/XenonRecomp/recompiler.cpp b/XenonRecomp/recompiler.cpp
index fa91527..2ca92ba 100644
--- a/XenonRecomp/recompiler.cpp
+++ b/XenonRecomp/recompiler.cpp
@@ -1629,8 +1629,16 @@ bool Recompiler::Recompile(
             print("{}.u32 + ", r(insn.operands[1]));
         println("{}.u32;", r(insn.operands[2]));

-        println("\tfor (size_t i = 0; i < ({} & 0xF); i++)", ea());
-        println("\t\tPPC_STORE_U8({} - i - 1, {}.u8[i]);", ea(), v(insn.operands[0]));
+        println("\t{");
+        println("\t\tuint32_t sh = {} & 0xF;", ea());
+        println("\t\tif (sh > 0) {");
+        println("\t\t\tsimde__m128i mask = simde_mm_load_si128((simde__m128i*)&VectorMaskR[sh * 16]);");
+        println("\t\t\tsimde__m128i data = simde_mm_shuffle_epi8(simde_mm_load_si128((simde__m128i*){}.u8), mask);", v(insn.operands[0]));
+        println("\t\t\tsimde__m128i mem = simde_mm_load_si128((simde__m128i*)(base + ({} & ~0xF)));", ea());
+        println("\t\t\tsimde__m128i blend_mask = simde_mm_cmpeq_epi8(mask, simde_mm_set1_epi8(char(0xFF)));");
+        println("\t\t\tsimde_mm_store_si128((simde__m128i*)(base + ({} & ~0xF)), simde_mm_blendv_epi8(data, mem, blend_mask));", ea());
+        println("\t\t}");
+        println("\t}");
         break;

     case PPC_INST_STVX:
diff --git a/XenonUtils/xbox.h b/XenonUtils/xbox.h
index 0361b0f..8763c18 100644
--- a/XenonUtils/xbox.h
+++ b/XenonUtils/xbox.h
@@ -29,15 +29,6 @@ struct be
 {
     T value;

-    be() : value(0)
-    {
-    }
-
-    be(const T v)
-    {
-        set(v);
-    }
-
     static T byteswap(T value)
     {
         if constexpr (std::is_same_v<T, double>)
@@ -101,12 +92,14 @@ struct xpointer
 {
     be<uint32_t> ptr;

-    xpointer() : ptr(0)
+    xpointer()
     {
+        ptr = 0;
     }

-    xpointer(T* p) : ptr(p != nullptr ? (reinterpret_cast<size_t>(p) - reinterpret_cast<size_t>(MmGetHostAddress(0))) : 0)
+    xpointer(T* p)
     {
+        ptr = (p != nullptr ? (reinterpret_cast<size_t>(p) - reinterpret_cast<size_t>(MmGetHostAddress(0))) : 0);
     }

     T* get() const
