diff --git a/XenonRecomp/recompiler.cpp b/XenonRecomp/recompiler.cpp
index fa91527..1b016e3 100644
--- a/XenonRecomp/recompiler.cpp
+++ b/XenonRecomp/recompiler.cpp
@@ -2004,35 +2004,28 @@ bool Recompiler::Recompile(
             if (insn.operands[3] != 1)
                 fmt::println("Unexpected D3D color pack instruction at {:X}", base);

-            for (size_t i = 0; i < 4; i++)
-            {
-                constexpr size_t indices[] = { 3, 0, 1, 2 };
-                println("\t{}.u32[{}] = 0x404000FF;", vTemp(), i);
-                println("\t{}.f32[{}] = {}.f32[{}] < 3.0f ? 3.0f : ({}.f32[{}] > {}.f32[{}] ? {}.f32[{}] : {}.f32[{}]);", vTemp(), i, v(insn.operands[1]), i, v(insn.operands[1]), i, vTemp(), i, vTemp(), i, v(insn.operands[1]), i);
-                println("\t{}.u32 {}= uint32_t({}.u8[{}]) << {};", temp(), i == 0 ? "" : "|", vTemp(), i * 4, indices[i] * 8);
-            }
-            println("\t{}.u32[{}] = {}.u32;", v(insn.operands[0]), insn.operands[4], temp());
+            println("\t{}.f32 = simde_mm_min_ps(simde_mm_max_ps({}.f32, simde_mm_set1_ps(3.0f)), simde_mm_castsi128_ps(simde_mm_set1_epi32(0x404000FF)));", vTemp(), v(insn.operands[1]));
+            println("\t{}.u32[{}] = simde_mm_cvtsi128_si32(simde_mm_shuffle_epi8(simde_mm_castps_si128({}.f32), simde_mm_set_epi8(char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(-1), char(0), char(12), char(8), char(4))));", v(insn.operands[0]), insn.operands[4], vTemp());
             break;

         case 5: // float16_4
             if (insn.operands[3] != 2 || insn.operands[4] > 2)
                 fmt::println("Unexpected float16_4 pack instruction at {:X}", base);

-            for (size_t i = 0; i < 4; i++)
+            println("\tsimde_mm_store_si128((simde__m128i*){}.u16, simde_mm_cvtps_ph(simde_mm_load_ps({}.f32), _MM_FROUND_TO_ZERO));", vTemp(), v(insn.operands[1]));
+            println("\tsimde_mm_store_si128((simde__m128i*){}.u16, simde_mm_or_si128(simde_mm_load_si128((simde__m128i*){}.u16), simde_mm_and_si128(simde_mm_cmpeq_epi16(simde_mm_and_si128(simde_mm_load_si128((simde__m128i*){}.u16), simde_mm_set1_epi16(short(0x7C00))), simde_mm_set1_epi16(short(0x7C00))), simde_mm_set1_epi16(short(0x03FF)))));", vTemp(), vTemp(), vTemp());
+
+            switch (insn.operands[4])
             {
-        		// Strip sign from source
-        		println("\t{}.u32 = ({}.u32[{}]&0x7FFFFFFF);", temp(), v(insn.operands[1]), i);
-        		// If |source| is > 65504, clamp output to 0x7FFF, else save 8 exponent bits
-        		println("\t{0}.u8[0] = ({1}.f32 != {1}.f32) || ({1}.f32 > 65504.0f) ? 0xFF : (({2}.u32[{3}]&0x7f800000)>>23);", vTemp(), temp(), v(insn.operands[1]), i);
-        		// If 8 exponent bits were saved, it can only be 0x8E at most
-        		// If saved, save first 10 bits of mantissa
-        		println("\t{}.u16 = {}.u8[0] != 0xFF ? (({}.u32[{}]&0x7FE000)>>13) : 0x0;", temp(), vTemp(), v(insn.operands[1]), i);
-        		// If saved and > 127-15, exponent is converted from 8 to 5-bit by subtracting 0x70
-        		// If saved but not > 127-15, clamp exponent at 0, add 0x400 to mantissa and shift right by (0x71-exponent)
-        		// If right shift is greater than 31 bits, manually clamp mantissa to 0 or else the output of the shift will be wrong
-        		println("\t{0}.u16[{1}] = {2}.u8[0] != 0xFF ? ({2}.u8[0] > 0x70 ? ((({2}.u8[0]-0x70)<<10)+{3}.u16) : (0x71-{2}.u8[0] > 31 ? 0x0 : ((0x400+{3}.u16)>>(0x71-{2}.u8[0])))) : 0x7FFF;", v(insn.operands[0]), i+(2*insn.operands[4]), vTemp(), temp());
-        		// Add back original sign
-        		println("\t{}.u16[{}] |= (({}.u32[{}]&0x80000000)>>16);", v(insn.operands[0]), i+(2*insn.operands[4]), v(insn.operands[1]), i);
+            case 0:
+                println("\tsimde_mm_store_si128((simde__m128i*){}.u16, simde_mm_blend_epi16(simde_mm_load_si128((simde__m128i*){}.u16), simde_mm_load_si128((simde__m128i*){}.u16), 0x0F));", v(insn.operands[0]), v(insn.operands[0]), vTemp());
+                break;
+            case 1:
+                println("\tsimde_mm_store_si128((simde__m128i*){}.u16, simde_mm_blend_epi16(simde_mm_load_si128((simde__m128i*){}.u16), simde_mm_slli_si128(simde_mm_load_si128((simde__m128i*){}.u16), 4), 0x3C));", v(insn.operands[0]), v(insn.operands[0]), vTemp());
+                break;
+            case 2:
+                println("\tsimde_mm_store_si128((simde__m128i*){}.u16, simde_mm_blend_epi16(simde_mm_load_si128((simde__m128i*){}.u16), simde_mm_slli_si128(simde_mm_load_si128((simde__m128i*){}.u16), 8), 0xF0));", v(insn.operands[0]), v(insn.operands[0]), vTemp());
+                break;
             }
             break;

diff --git a/XenonUtils/ppc_context.h b/XenonUtils/ppc_context.h
index 31a86cd..a3332bd 100644
--- a/XenonUtils/ppc_context.h
+++ b/XenonUtils/ppc_context.h
@@ -15,6 +15,7 @@
 #include <x86/avx.h>
 #include <x86/sse.h>
 #include <x86/sse4.1.h>
+#include <x86/f16c.h>

 // SSE3 constants are missing from simde
 #ifndef _MM_DENORMALS_ZERO_MASK
diff --git a/XenonUtils/xbox.h b/XenonUtils/xbox.h
index 0361b0f..215878d 100644
--- a/XenonUtils/xbox.h
+++ b/XenonUtils/xbox.h
@@ -247,13 +247,13 @@ typedef struct _XXOVERLAPPED {
         {
             be<uint32_t> Error;
             be<uint32_t> Length;
-        };
+        } AsError;

         struct
         {
             uint32_t InternalLow;
             uint32_t InternalHigh;
-        };
+        } AsInternal;
     };
     uint32_t InternalContext;
     be<uint32_t> hEvent;
diff --git a/XenonUtils/xdbf.h b/XenonUtils/xdbf.h
index 347ac2c..b9e5b87 100644
--- a/XenonUtils/xdbf.h
+++ b/XenonUtils/xdbf.h
@@ -132,7 +132,7 @@ union XDBFTitleID
     {
         be<uint16_t> u16;
         char u8[0x02];
-    };
+    } AsU16;

     be<uint32_t> u32;
 };
